python/semantic_kernel/functions/kernel_function.py

```diff
@@ -461,11 +441,11 @@ def gather_function_parameters(self, kernel: "Kernel", arguments: "KernelArgumen
             if param.name == "kernel":
                 function_arguments[param.name] = kernel
                 continue
-            if param.name == "client":
-                function_arguments[param.name] = self.ai_service
+            if param.name == "service":
+                function_arguments[param.name] = kernel.select_ai_service(self, arguments)[0]
```

Comment:
Moved it here because some functions don't use it, guess there is a different way to do it but this worked and the function isn't heavy, but I guess someone can override with something more complex... I'll have a look.